### DISCUSSION TITLE
add prepare prop to `NumberInput` component

### DISCUSSION THREAD

**Discussion by mohammadizanloo55** [mantinedev/mantine#5475] [Feature requests]
# Enhancing `NumberInput` Component to Support Persian and Arabic Numerals

Dear Mantine Development Team,

I am writing to propose an enhancement to the `NumberInput` component in the Mantine library, aimed at improving user experience for applications targeting Iranian and Arab users.

## Description:
Currently, the `NumberInput` component exclusively supports English numerals. This limitation poses a usability challenge for users with Persian or Arabic keyboard layouts, who naturally input numbers in their native script (e.g., ۱۲۳). 

**Proposed Solution:**
To address this, I suggest implementing a functionality similar to the [`react-imask`](https://imask.js.org/guide.html#masked-base) library, where developers can easily convert Persian and Arabic numerals into English numerals. This conversion would take place within the `NumberInput` component, seamlessly translating user input into a format compatible with most backend systems and data processing logic.

Here is a conceptual example to illustrate the proposed enhancement:

```javascript
<NumberInput
    prepare={(inputValue) => {
        return normalizeArabicAndPersianNumbers(inputValue);
    }}
    label="Input label"
    onChange={(value) => {
        // Output value is automatically converted using the prepare function
        console.log(value);
    }}
    description="Input description"
    placeholder="Input placeholder"
/>
```

**Benefits:**
- **Enhanced Accessibility:** Users with Persian or Arabic keyboard layouts can input numbers without having to switch to an English layout.
- **Improved User Experience:** Streamlines data entry for a significant user base, reducing friction and enhancing overall usability.
- **Broader Appeal:** This feature would make Mantine more attractive to developers creating applications for Middle Eastern markets.

I believe this enhancement will significantly benefit the developer community and end-users alike by accommodating a more diverse user base.

Thank you for considering this proposal. I am looking forward to any thoughts or feedback you might have on this matter.

**Comment by rtivital** [comment]
Thanks for the proposal! NumberInput uses [react-number-format](https://github.com/s-yadav/react-number-format) library. I suppose that it will be difficult to implement and hard to maintain.
